### PR TITLE
Implementation of GitPython instead of calling subprocess

### LINKED ISSUE #178: Replace git-wrapping code with new or existing library
Investigate existing libraries (example https://github.com/release-depot/git_wrapper).
Decide whether to use an existing library or move our [git-related code](https://github.com/packit-service/ogr/blob/master/ogr/utils.py) (check for all occurrences of "git" through the code) into a new library.
Use such library in ogr.

### PR DESCRIPTION
**Pull request by Hojang2**
This PR contains possible solution of #178 
Closes #178

### CODE REVIEW

**Comment by Hojang2** [comment]
Removed unused CLONE_TIMEOUT constant.

**Comment by lachmanfrantisek** [comment]
> * [ogr-pre-commit ](https://softwarefactory-project.io/zuul/t/local/build/898a9349683147dcae4da94c10dfce1b) : FAILURE in 4m 07s

The check failed because you need to [rebase](https://git-scm.com/book/en/v2/Git-Branching-Rebasing) your branch on top of the current master.

  - We want to have a git tree like this: 
Snímek z 2019-09-19 11-43-15
    - not this: 
Snímek z 2019-09-19 11-43-51

**Comment by lachmanfrantisek** [comment]
```
black....................................................................Failed
hookid: black

Files were modified by this hook. Additional output:

reformatted /home/zuul-worker/src/github.com/packit-service/ogr/ogr/utils.py
All done! ✨ 🍰 ✨
1 file reformatted, 43 files left unchanged.
```

Look at the info about our `pre-commit` setup in [our CONTRIBUTION guide](https://github.com/packit-service/ogr/blob/master/CONTRIBUTING.md#checkerslintersformatters--pre-commit) -- we are using [black](https://github.com/ambv/black) for formatting our code. Pre-commit can help you with that and reformat your code before each commit.

**Comment by Hojang2** [comment]
@TomasTomecek I do not understand what do you mean by "/packit copr-build". I am new to packit and ogr.

**Comment by Hojang2** [comment]
> /packit copr-build

**Comment by TomasTomecek** [comment]
@Hojang2 `/packit copr-build` is a command to trigger an rpm build of ogr - we do that usually when something goes wrong and we need to retrigger the build

sadly it's not documented in the contribution guidelines - will open an issue for that

File: ogr/utils.py [hunk 1]
```diff
@@ -25,42 +25,45 @@
...
 
-    proc = subprocess.run(
-        ["git", "clone", repo_ssh_url], stderr=subprocess.PIPE, timeout=CLONE_TIMEOUT
-    )
-    output = proc.stderr.decode()
-    logger.debug("Clone exited with {} and output: {}".format(proc.returncode, output))
-    if "does not exist" in output:
-        logger.error("Clone failed.")
-        raise Exception("Clone failed")
+    if not os.path.exists(namespace):
```
Comment (TomasTomecek) [inline ogr/utils.py]: what's the reason for this if, sounds pretty odd to be honest

**Pull request merged**